app/src/main/java/com/nutomic/syncthingandroid/activities/FirstStartActivity.java

```diff
@@ -230,26 +300,42 @@ public Object instantiateItem(ViewGroup container, int position) {
 
             View view = layoutInflater.inflate(mLayouts[position], container, false);
 
-            /* Slide: storage permission */
-            Button btnGrantStoragePerm = (Button) view.findViewById(R.id.btnGrantStoragePerm);
-            if (btnGrantStoragePerm != null) {
-                btnGrantStoragePerm.setOnClickListener(new View.OnClickListener() {
-                    @Override
-                    public void onClick(View v) {
-                        requestStoragePermission();
-                    }
-                });
-            }
+            switch (slideOrder[position]) {
+                case INTRO:
+                    break;
+
+                case STORAGE:
+                    Button btnGrantStoragePerm = (Button) view.findViewById(R.id.btnGrantStoragePerm);
+                    btnGrantStoragePerm.setOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            requestStoragePermission();
+                        }
+                    });
+                    break;
 
-            /* Slide: location permission */
-            Button btnGrantLocationPerm = (Button) view.findViewById(R.id.btnGrantLocationPerm);
-            if (btnGrantLocationPerm != null) {
-                btnGrantLocationPerm.setOnClickListener(new View.OnClickListener() {
-                    @Override
-                    public void onClick(View v) {
-                        requestLocationPermission();
-                    }
-                });
+                case LOCATION:
+                    Button btnGrantLocationPerm = (Button) view.findViewById(R.id.btnGrantLocationPerm);
+                    btnGrantLocationPerm.setOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            requestLocationPermission();
+                        }
+                    });
+                    break;
+
+                case API_LEVEL_30:
+                    Button btnResetDatabase = (Button) view.findViewById(R.id.btnResetDatabase);
+                    btnResetDatabase.setOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            // The main-activity will actually do the db reset.
+                            mResetDatabase = true;
+                            mPreferences.edit().putBoolean(Constants.PREF_UPGRADED_TO_API_LEVEL_30, true).apply();
+                            onBtnNextClick();
```

Comment:
What if I kill the app after this slide, or it ends up with syncthing being disabled due not being connected to wifi or something, etc, I exit, come back, come back later, it then it wouldn't set the flag and potentially blow up in my face.

Is there any harm in calling `syncthing -reset-database` or something of that like right here?
Also wonder if it's not just easier to `rm -rf` the database directory oppose to relying on actually launching the binary etc.

Feels like the flag should only be set after the actual launch that did the reset.

Reply:
There was some problem with starting syncthing here, and not in the main activity. Already forgot again what it was. I'll set the flag only after that ran successfully.